wpsc-includes/variations.class.php

```diff
@@ -296,3 +296,50 @@ function wpsc_get_child_object_in_terms_var( $parent_id, $terms, $taxonomies, $a
 		return false;
 	}
 }
+
+/**
+ * Given array of variation selections this works through the terms and returns the product_id for the matching variation
+ *
+ * @since 4.0
+ *
+ * @param array     $variations     required            The array of variation selections
```

Comment:
Let's change this function to `wpsc_get_product_data_from_variations`.

Then, let's have two wrapper functions for it, `wpsc_get_product_id_from_variations` and `wpsc_get_variation_values_from_variations`.

Finally, let's document clearly the expectations for the `$variations` parameter.  If the goal is to have this function abstracted to be useful elsewhere, we need to clearly describe the array of data intended to be passed.